Sources/T2ScholaCoreSwift/Request/PopupNotificationRequest.swift

```diff
@@ -0,0 +1,36 @@
+import Foundation
+#if canImport(FoundationNetworking)
+import FoundationNetworking
+#endif
+
+struct PopupNotificationRequest: RestAPIRequest {
+    typealias RequestBody = Void
+    typealias Response = PopupNotificationResponse
+    
+    let method: HTTPMethod = .get
+    
+    let queryParameters: [String: Any]?
+    
+    init(userId: Int, wsToken: String) {
+        queryParameters = [
+            "moodlewsrestformat" : "json",
+            "wstoken" : wsToken,
+            "useridto" : userId,
+            "wsfunction" : "message_popup_get_popup_notifications"
+        ]
+    }
+}
+
+public struct PopupNotificationResponse: Codable {
+    public let unreadcount: Int
+    public let notifications: [PopupNotification]
+}
+
+public struct PopupNotification: Codable {
+    public let subject: String
+    public let fullmessage: String
+    public let timecreated: Int
+    public let read: Bool
+    public let contexturl: String
```

Comment:
```
export type AddonNotificationsPopupNotification = {
    id: number; // Notification id (this is not guaranteed to be unique within this result set).
    useridfrom: number; // User from id.
    useridto: number; // User to id.
    subject: string; // The notification subject.
    shortenedsubject: string; // The notification subject shortened with ellipsis.
    text: string; // The message text formated.
    fullmessage: string; // The message.
    fullmessageformat: number; // Fullmessage format (1 = HTML, 0 = MOODLE, 2 = PLAIN or 4 = MARKDOWN).
    fullmessagehtml: string; // The message in html.
    smallmessage: string; // The shorten message.
    contexturl: string; // Context URL.
    contexturlname: string; // Context URL link name.
    timecreated: number; // Time created.
    timecreatedpretty: string; // Time created in a pretty format.
    timeread: number; // Time read.
    read: boolean; // Notification read status.
    deleted: boolean; // Notification deletion status.
    iconurl: string; // URL for notification icon.
    component?: string; // The component that generated the notification.
    eventtype?: string; // The type of notification.
    customdata?: string; // @since 3.7. Custom data to be passed to the message processor.
};
```